File: src/effs/get_env.c
// Console.get_env — copy an environment value into a Bend string.
// Missing variables return an empty string; the getenv pointer never escapes.

Term get_env_run(Env e, Term* f, IoWork* w) {
  uint64_t n = 0;
  char* name = io_cstr(e, f[0], &n);
  const char* value = getenv(name);
  Term result = io_done(e, io_str(e, value != NULL ? value : "", value != NULL ? strlen(value) : 0));
  free(name);
  (void)w;
  return result;
}

static void __attribute__((constructor)) get_env_use(void) {
  io_eff(CID_GET_ENV, get_env_run, 0);
}

File: src/effs/get_env.js
// Console.get_env — copy an environment value, mirroring get_env.c.

function get_env(name) {
  try {
    const key = Buffer.from(io_bytes(name)).toString("utf8");
    return io_done(String(process.env[key] ?? ""));
  } catch (e) {
    return io_fail(Math.abs(e.errno ?? 5));
  }
}

io_eff(CID(get_env), get_env);
